src/lib/network/wicked.rb

```diff
@@ -41,15 +41,38 @@ def reload_config(devs)
     # @param iface [String] network device
     # @return [Array<String>] list of NTP servers
     def parse_ntp_servers(iface)
+      query_wicked(iface, "%{//ntp/server}")
+    end
+
+    # Parses wicked runtime configuration and returns hostname if set
+    #
+    # @param iface [String] network device
+    # @return [String] hostname
+    def parse_hostname(iface)
+      result = query_wicked(iface, "%{//hostname}")
+
+      raise "Malformed wicked runtime configuration" if result.count > 1
+
+      result.first
+    end
+
+    # Parses wicked runtime dhcp lease file for the given query
+    #
+    # It parses both ipv4 and ipv6 lease files at once.
+    #
+    # @param iface [String] queried interface
+    # @param query [String] xpath query
```

Comment:
maybe mention that it is just subset of xpath ( as specified in man wicked section xpath )

Reply:
ok, I'll point to the wicked doc.